gulpfile.babel.js

```diff
@@ -1,112 +1,105 @@
   }
 };
-const getOptions = (opts = {}) =>
-  assign({}, jadeDefaults, opts, {
-    locals: assign({}, jadeDefaults.locals, opts.locals)
-  });
-
-const jade = opts => gulpJade(getOptions(opts));
 
-var articleData = require('./article-data');
+const getOptions = (opts = {}) => Object.assign({}, jadeDefaults, opts, {
+  locals: Object.assign({}, jadeDefaults.locals, opts.locals)
+});
 
-var articles = [];
-var articleHarvesting = function() {
-  return through.obj(function(file, enc, cb) {
-    var article = articleData(file.contents.toString());
-    var url = getBasename(file);
-    var title = article.title;
+const jade = opts => gulpJade(getOptions(opts));
 
-    if (getBasename(file) === 'README') {
-      url = 'about';
-      title = 'О проекте';
-    }
+let articles = [];
 
-    var author = fs.readJsonSync('./dump/' + title + '.json');
+const articleHarvesting = () => {
+  return through.obj((file, enc, cb) => {
+    const article = articleData(file.contents.toString());
+    const url = getBasename(file);
+    const title = article.title;
+    const author = fs.readJsonSync('./dump/' + title + '.json');
+    const firstTweet = author.tweets[author.tweets.length - 1];
+    const isReadme = getBasename(file) === 'README';
 
     articles.push({
       site: site,
       filename: file.relative,
-      url: url + '/',
-      title: title,
+      url: isReadme ? 'about/' : url + '/',
+      title: isReadme ? 'О проекте' : title,
       image: article.image,
-      desc: html(renderTweet(author.tweets[author.tweets.length - 1])),
-      descText: author.tweets[author.tweets.length - 1].text,
-      date: d(author.tweets[author.tweets.length - 1].created_at),
+      desc: html(renderTweet(firstTweet)),
+      descText: firstTweet.text,
+      date: d(firstTweet.created_at),
       content: article.content,
       latestInfo: latestInfo,
       rss: {
         url: site.site_url + getBasename(file) + '/',
-        description: author.tweets[author.tweets.length - 1].text
+        description: firstTweet.text
       }
     });
-    articles.sort(function(a, b) { return unix(b.date) - unix(a.date); });
+    articles.sort((a, b) => unix(b.date) - unix(a.date));
     cb(null, file);
   });
 };
 
-gulp.task('articles-registry', function() {
+gulp.task('articles-registry', () => {
   articles = [];
   return gulp.src(['./posts/*.md'])
     .pipe(replace('https://jsunderhood.ru', 'http://localhost:4000'))
     .pipe(articleHarvesting());
 });
 
-gulp.task('articles-registry-prod', function() {
+gulp.task('articles-registry-prod', () => {
   articles = [];
   return gulp.src(['./posts/*.md', '!./posts/*draft*.md'])
     .pipe(articleHarvesting());
 });
 
-gulp.task('index-page', function() {
-  var groupInGrid = function(state, item, i) {
+gulp.task('index-page', () => {
+  const groupInGrid = (state, item, i) => {
     if (i % 3 === 0) { state.push([]); }
     state[state.length - 1].push(item);
     return state;
```

Comment:
what do you think about extracting this function to out scope, and then just using it here